controllers/storagecluster/cephcluster.go

```diff
@@ -1256,3 +1264,31 @@ func getIPFamilyConfig(c client.Client) (rookCephv1.IPFamilyType, bool, error) {
 
 	return rookCephv1.IPv4, false, nil
 }
+
+func getOSDStoreConfig(sc *ocsv1.StorageCluster) rookCephv1.OSDStore {
+	osdStore := rookCephv1.OSDStore{}
+	if !sc.Spec.ExternalStorage.Enable && optimizeDisasterRecovery(sc) {
+		osdStore.Type = string(rookCephv1.StoreTypeBlueStoreRDR)
+	}
+
+	return osdStore
+}
+
+// optimizeDisasterRecovery returns true if any RDR optimizations are required
+func optimizeDisasterRecovery(sc *ocsv1.StorageCluster) bool {
+	if annotation, found := sc.GetAnnotations()[DROptimizationAnnotation]; found {
+		if annotation == "true" {
+			return true
+		}
+	}
+
+	return false
+}
+
+func updateRDROptimizations(newOSDStore, existingOSDStore rookCephv1.OSDStore) rookCephv1.OSDStore {
```

Comment:
How about this method name?
```suggestion
func determineOSDStore(newOSDStore, existingOSDStore rookCephv1.OSDStore) rookCephv1.OSDStore {
```